### PR TITLE
simplify our run name, lets just call it run-ci

### PR DESCRIPTION
**Pull request by vasukulkarni**
simplify our run name, lets just call it run-ci (the repo name is ocs-ci, so it would imply
running ocs-ci)

Signed-off-by: Vasu Kulkarni <vasu@redhat.com>

### CODE REVIEW

**Comment by vasukulkarni** [comment]
> I think you'll still need to type 'python -m run-ci'.

entry-point binaries are created during venv setup so we dont have to add python.
```
$ ls venv/bin/run-ci 
venv/bin/run-ci
```

**Comment by petr-balogh** [comment]
> > I think you'll still need to type 'python -m run-ci'.
> 
> entry-point binaries are created during venv setup so we dont have to add python.
> 
> ```
> $ ls venv/bin/run-ci 
> venv/bin/run-ci
> ```

Yup didn't realized this yesterday late night, thanks for bringing this up ;)

Can you also try to change this in [tox.ini](https://github.com/red-hat-storage/ocs-ci/blob/master/tox.ini) file? Than we will that in travis that this really works as well.

Thanks

**Comment by RazTamir** [comment]
It still doesn't look right:
before, it used to be 'python -m run_ocsci -m deployment ...' and you changed it to 'run-ci -m deployment ...'.
In setup.py the change is:
```python
entry_points={
        'console_scripts': [
---            'run_ocsci=run_ocsci:main',
+++         'run-ci=run_ocsci:main',
        ],
```

So IIUC, you just changed the name of the entry point but python -m still needed.
Sorry if I miss something very simple

**Comment by petr-balogh** [comment]
> It still doesn't look right:
> before, it used to be 'python -m run_ocsci -m deployment ...' and you changed it to 'run-ci -m deployment ...'.
> In setup.py the change is:
> 
> ```python
> entry_points={
>         'console_scripts': [
> ---            'run_ocsci=run_ocsci:main',
> +++         'run-ci=run_ocsci:main',
>         ],
> ```
> 
> So IIUC, you just changed the name of the entry point but python -m still needed.
> Sorry if I miss something very simple
@RazTamir did you reinstall venv with pip install -r requirements.txt  ?

**Comment by RazTamir** [comment]
yes

**Comment by petr-balogh** [comment]
> yes

I've just tested on one of jenkins machine with removing .venv and created new one and looks running ok:

```bash
(.venv) [jenkins@qe-ocs4-slave-prod-3 ocs-ci]$ run-ci -m deployment --cluster-name pbalogh-run --cluster-path /home/jenkins/testing-run-ci
The kubeconfig file /home/jenkins/testing-run-ci/auth/kubeconfig doesn't exist!
================================================ test session starts ================================================
platform linux -- Python 3.6.6, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /home/jenkins/ocs-ci/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/jenkins/ocs-ci, inifile: pytest.ini
plugins: ocs-ci-0.1
collected 6 items / 5 deselected / 1 selected                                                                       

tests/ecosystem/deployment/test_ocs_basic_install.py::TestDeployment::test_deployment 
--------------------------------------------------- live log call ---------------------------------------------------
17:41:09 - MainThread - tests.ecosystem.deployment.test_ocs_basic_install - INFO - Running OCS basic installation
17:41:09 - MainThread - oc.openshift_ops - INFO - Testing access to cluster with /home/jenkins/testing-run-ci/auth/kubeconfig
17:41:09 - MainThread - oc.openshift_ops - WARNING - The kubeconfig file /home/jenkins/testing-run-ci/auth/kubeconfig doesn't exist!
17:41:09 - MainThread - tests.ecosystem.deployment.test_ocs_basic_install - INFO - Generating install-config
17:41:09 - MainThread - utility.utils - INFO - Executing command: mkdir -p /home/jenkins/testing-run-ci
17:41:09 - MainThread - tests.ecosystem.deployment.test_ocs_basic_install - INFO - Install config: 
```

Just aborted it right after

**Comment by vasukulkarni** [comment]
> It still doesn't look right:
> before, it used to be 'python -m run_ocsci -m deployment ...' and you changed it to 'run-ci -m deployment ...'.
> In setup.py the change is:
> 
> ```python
> entry_points={
>         'console_scripts': [
>             'run_ocsci=run_ocsci:main',  #  Removed
>             'run-ci=run_ocsci:main',  #  Added
>         ],
> ```
> 
> So IIUC, you just changed the name of the entry point but python -m still needed.
> Sorry if I miss something very simple

The python's -m option is for running main function of the module, but since we are generating the binary, we can directly call it and the -m option is now py.test's mark option. so 'run-ci -m' =~ 'py.test -m' with preprocessing.

**Comment by RazTamir** [comment]
Thanks for the explanation,
I've learned something new (:

**Comment by petr-balogh** [comment]
@vasukulkarni please take a look to my comment where I am talking about tox.ini file. Change looks OK, but will be good to have it also there to be sure it's working as expected and cover by unittests as well. Thanks

**Comment by vasukulkarni** [comment]
> > > I think you'll still need to type 'python -m run-ci'.
> > 
> > 
> > entry-point binaries are created during venv setup so we dont have to add python.
> > ```
> > $ ls venv/bin/run-ci 
> > venv/bin/run-ci
> > ```
> 
> Yup didn't realized this yesterday late night, thanks for bringing this up ;)
> 
> Can you also try to change this in [tox.ini](https://github.com/red-hat-storage/ocs-ci/blob/master/tox.ini) file? Than we will that in travis that this really works as well.
> 
> Thanks

Sure thanks for reminding.

**Comment by RazTamir** [comment]
Could you please send some update to ocs-qe about this?
And the first question about this change, do I need to execute pip install -r requirements.txt?

**Comment by clacroix12** [comment]
@RazTamir yes you will still need to follow the `getting_started` doc which includes pip installing our requirements.

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,9 @@ envlist = py37,flake8
 [testenv]
 deps =
     -rrequirements.txt
-commands = {envpython} -m run_ocsci --ignore=tests -c unittests/pytest.ini {posargs:unittests}
+setenv =
+ PYTHONPATH=.
```
Comment (mbukatov) [inline tox.ini]: Why is this a good idea exactly?

**Comment by mbukatov** [comment]
I vote to revert this: https://github.com/red-hat-storage/ocs-ci/pull/142/files

**Comment by vasukulkarni** [comment]
@mbukatov entry-points are well documented, I suggest you go ahead and read how it works.

**Comment by mbukatov** [comment]
@vasukulkarni I know how entripoints work. But using entripoint wrappers which references local module is not expected usage pattern. The problem is in pythonpath and visibility of the run_ocsci module.

**Comment by vasukulkarni** [comment]
> @vasukulkarni I know how entripoints work. But using entripoint wrappers which references local module is not expected usage pattern. The problem is in pythonpath and visibility of the run_ocsci module.

It works in venv, I am not sure why its not working for you and you are *claiming* the docs is wrong, If you want to ship the package thats a different story.

```
(venv2) Vasus-MacBook-Pro:ocs-ci vasukulkarni$ echo $PYTHONPATH

(venv2) Vasus-MacBook-Pro:ocs-ci vasukulkarni$ run-ci --help
usage: run-ci [options] [file_or_dir] [file_or_dir] [...]

positional arguments:
  file_or_dir

general:
  -k EXPRESSION         only run tests which match the given substring
                        expression. An expression is a python evaluatable
                        expression where all names are substring-matched
                        against test names and their parent classes. Example:
                        -k 'test_method or test_other' matches all test
                        functions and classes whose name contains
                        'test_method' or 'test_other', while -k 'not
                        test_method' matches those that don't contain
                        'test_method' in their names. -k 'not test_method and
                        not test_other' will eliminate the matches.
                        Additionally keywords are matched to classes and
                        functions containing extra names in their
                        'extra_keyword_matches' set, as well as functions
                        which have names assigned directly to them.
  -m MARKEXPR           only run tests matching given mark expression.
                        example: -m 'mark1 and not mark2'.
  --markers             show markers (builtin, plugin and per-project ones).

```

**Pull request merged**